### PR TITLE
add emodels relationship to ion channel models

### PR DESCRIPTION
**Pull request by AurelienJaquier**
This should solve the issue raised in this comment: https://github.com/openbraininstitute/prod-explore-functionality/issues/142#issuecomment-3345950795

### CODE REVIEW

**Comment by AurelienJaquier** [comment]
@g-bar I think this should solve the problem we had in [your comment](https://github.com/openbraininstitute/prod-explore-functionality/issues/142#issuecomment-3345950795). Could you tell me if this PR would indeed solve it with a review?

**Comment by g-bar** [comment]
I think this would work for me but I’ll let @GianlucaFicarelli and @eleftherioszisis review it.

File: app/service/ion_channel_model.py [hunk 1]
```diff
@@ -55,6 +55,7 @@ def _load_expanded(q: Select[IonChannelModel]) -> Select[IonChannelModel]:
         selectinload(IonChannelModel.contributions).selectinload(Contribution.agent),
         selectinload(IonChannelModel.contributions).selectinload(Contribution.role),
         selectinload(IonChannelModel.assets),
+        selectinload(IonChannelModel.emodels),
```
Comment (eleftherioszisis) [inline app/service/ion_channel_model.py:58]: I understand EModel having a has-a relationship of ion-channel models, but the opposite isn't strange? This seems like a case of querying, not explicit relationship. In addition, this introduces a cycle. Does this really work? I am surprised that this would work without joining the relationships of EModel.

In any case, please add tests to ensure that the functionality that is added really works and it isn't just lazy loading or empty records that let tests pass without complaints.

**Review by eleftherioszisis — CHANGES_REQUESTED** [review]

File: app/service/ion_channel_model.py (hunk 1 above)
Comment (AurelienJaquier) [reply app/service/ion_channel_model.py:58]: For the [display of ion channel model in explore](https://github.com/openbraininstitute/prod-explore-functionality/issues/142) we need to get all emodels that use the ion channel.
If we want to use querying instead of relationship, what should we have to change in entitycore / entitysdk?
